commit 02a9a39294017f105aedebcca5f49d552b18dbaa
Author: Jamie Lentin <jm@lentin.co.uk>
Date:   Tue Apr 7 22:59:56 2009 +0100

    at76c50x-usb: Add device ID for OQO model 01+
    
    Add USB device ID for OQO 01+'s internal wireless LAN
    
    An OQO employee mentions the chip's true identity here:-
        ftp://ftp.oqo.com/unsupported/linux/OQOLinux.html
    
    Signed-off-by: Jamie Lentin <jm@lentin.co.uk>
    Acked-by: Kalle Valo <kalle.valo@iki.fi>
    Signed-off-by: John W. Linville <linville@tuxdriver.com>

diff --git a/drivers/net/wireless/at76c50x-usb.c b/drivers/net/wireless/at76c50x-usb.c
index 0c02f1c..744f4f4 100644
--- a/drivers/net/wireless/at76c50x-usb.c
+++ b/drivers/net/wireless/at76c50x-usb.c
@@ -250,6 +250,8 @@ static struct usb_device_id dev_table[] = {
 	{ USB_DEVICE(0x03eb, 0x7617), USB_DEVICE_DATA(BOARD_505A) },
 	/* Siemens Gigaset USB WLAN Adapter 11 */
 	{ USB_DEVICE(0x1690, 0x0701), USB_DEVICE_DATA(BOARD_505A) },
+	/* OQO Model 01+ Internal Wi-Fi */
+	{ USB_DEVICE(0x1557, 0x0002), USB_DEVICE_DATA(BOARD_505A) },
 	/*
 	 * at76c505amx-rfmd
 	 */
